src/System.Net.Security/src/System/Net/Security/SecureChannel.cs

```diff
@@ -839,13 +861,16 @@ private SecurityStatusPal GenerateToken(byte[] input, int offset, int count, ref
                     //
                     if (!cachedCreds && _securityContext != null && !_securityContext.IsInvalid && _credentialsHandle != null && !_credentialsHandle.IsInvalid)
                     {
-                        SslSessionsCache.CacheCredential(_credentialsHandle, thumbPrint, _sslProtocols, _serverMode, _encryptionPolicy);
+                        SslSessionsCache.CacheCredential(_credentialsHandle, thumbPrint, _sslAuthenticationOptions.EnabledSslProtocols, _sslAuthenticationOptions.IsServer, _sslAuthenticationOptions.EncryptionPolicy);
```

Comment:
Should the cache/fetch just take all the entire options object and key off of what makes sense inside?